cli/tsc/diagnostics.rs

```diff
@@ -273,6 +280,51 @@ impl Diagnostics {
   pub fn is_empty(&self) -> bool {
     self.0.is_empty()
   }
+
+  /// Modifies all the diagnostics to have their display positions
+  /// modified to point at the original source.
+  pub fn apply_fast_check_source_maps(&mut self, graph: &ModuleGraph) {
+    fn visit_diagnostic(d: &mut Diagnostic, graph: &ModuleGraph) {
+      if let Some(specifier) = d
+        .file_name
+        .as_ref()
+        .and_then(|n| ModuleSpecifier::parse(n).ok())
+      {
+        if let Ok(Some(module)) = graph.try_get_prefer_types(&specifier) {
+          if let Some(fast_check_module) =
+            module.esm().and_then(|m| m.fast_check_module())
+          {
+            // todo(dsherret): use a short lived cache to prevent parsing
+            // source maps so often
+            if let Ok(source_map) =
+              SourceMap::from_slice(&fast_check_module.source_map)
+            {
```

Comment:
Could you do it in this PR by adding a cache in `apply_fast_check_source_map`? Or do you feel it's not gonna have much impact without it for now?

Reply:
I opened https://github.com/denoland/deno/issues/21877